### ISSUE TITLE
Data Table: Inline Editing

### ISSUE THREAD

**Issue by joshblack** [IBM/carbon-components-react#489]
This issue is to add in the inline editing behavior for `DataTable` and includes the following:

- [ ] Create initial API design for Inline editing
- [ ] Gather feedback from sponsor teams on API design
- [ ] Initial implementation of inline editing
- [ ] Create corresponding storybook story for feature
- [ ] Add corresponding tests for feature
- [ ] Conduct self-audit on any potential a11y or i18n hiccups we might run into

[Spec](https://github.ibm.com/Bluemix/carbon-issues/issues/385)

**Comment by joshblack** [comment]
Closing in favor of: https://ibm.biz/BdZnS7

**Comment by zacksinclair** [comment]
@joshblack I'm interested in this feature, but can't get to the ibm-only link to see the latest details. Is there a plan to add inline editing in the near future?

**Comment by joshblack** [comment]
Hi @zacksinclair! 👋 Sorry for the delay. Let me go ahead and re-open this issue.

We would love to add inline editing, when it comes to roadmap it currently isn't a priority as we are currently on the v10 style (preview: https://next.carbondesignsystem.com/). We 100% want to add this in though! Most likely this effort would be two-fold:

- Add the styles for inline editing to `carbon-components`
- Add the rough implementation to `carbon-components-react`

Part of the reason we delayed work here is that there was a belief that some newer React features like suspense and hooks would help us build out a better API to manage this work than us trying to implement this with the current tooling.

**Comment by asudoh** [comment]
Weighing in - Data table stuff tends to involve a lot of application-specific logic and our UI component code attempting to do dictate every aspect has yielded lots of issues coming back to us. Application should have a choice on usage of suspense/hooks API, especially given where those API would be applied to is much closer to application side rather than UI side.

**Comment by zacksinclair** [comment]
Fair enough guys, thank you for the responses. Seems like waiting it out in favor of suspense/hooks is the right choice for now. Appreciate your work on this project!
